### ISSUE TITLE
GIT_FULL_HASH is undefined error on conda build

### ISSUE THREAD

**Issue by betranjacob** [conda/conda-build#1846]
I am getting a "GIT_FULL_HASH is undefined" error when running conda build . I can get the hash manually by issuing the command - "git rev-parse HEAD"
Is there anything specific I should configure to set these variables? 

package:
    name: myFirstModule
    # String with the full SHA1 of the current HEAD
    version: {{ GIT_FULL_HASH }}

**Comment by msarahan** [comment]
Do you have a source section in your recipe?  Conda-build only supports GIT_* variables if you have

```
source:
  git_url: (https url of repo, or relative path to repo on local filesystem.  relative to where meta.yaml lives)
```

or

```
source:
  path: (some relative filesystem path - relative to where meta.yaml for recipe is)
```

**Comment by betranjacob** [comment]
Yes, the source path is set to parent directory.
source:
  path: ..

My directory structure is as below - 
MyPacakge
   -setup.py
   -myFirstModule
   -condarecipe
             -meta.yaml
            - build.sh
 
 build command - "conda build condarecipe"

**Comment by msarahan** [comment]
That all seems in order.  Can you push this git repo somewhere?  I'll be happy to clone it and try to debug it.

**Comment by betranjacob** [comment]
Hi,  I have pushed the repo here :  https://github.com/betranjacob/conda_build_test
Please have a look.

**Comment by msarahan** [comment]
@betranjacob sorry for the delay.  #1873 should fix this issue.  I recommend also changing your pinnings.  In particular, meta.yaml does not support pins in this format:  ``==3.5``.  I'll put up a PR against your repository.

**Comment by msarahan** [comment]
Please see suggestions at https://gist.github.com/msarahan/91449694453107453ebc99b5d10afaf8

Basically, it is better to skip configurations that don't match your python version than it is to specify the version of python in the constraints.  This is just an implementation detail of conda-build.  It defaults to using the same Python as your miniconda/anaconda installation, and you can change it with the --python flag:

``conda build condarecipe --python=3.5``

This prevents some odd errors where conda-build will install the wrong version of python, and you'll end up with fairly confusing errors.

**Comment by msarahan** [comment]
fixed by #1873 - closing.

**Comment by betranjacob** [comment]
Hi, thanks for the fix. May I know when will this available on an official version?
